service/history/workflow/transaction_impl.go

```diff
@@ -23,6 +25,9 @@ type (
 		namespaceState   string
 		workflowTypeName string
 		status           enumspb.WorkflowExecutionStatus
+		state            enumsspb.WorkflowExecutionState
```

Comment:
not sure why we need this. `Status` alone can tell if workflow is closed or not.